### ISSUE TITLE
problems when newline `\n\n` is in the string. 

### ISSUE THREAD

**Issue by inteli5** [houfu/redlines#2]
Here is the code that shows the problem.
```python
from redlines import Redlines
import markdown
from IPython.display import display, Markdown,  HTML
test_string_1 = '''
Happy Saturday,

Thank you for reaching out, have a good weekend

Sophia 
'''
test_string_2 = '''Happy Saturday,

Thank you for reaching out. Have a good weekend.

Sophia.'''

diff = Redlines(test_string_1 ,test_string_2, markdown_style="none")
html=markdown.markdown(diff.output_markdown)
display(HTML(html))
```

When displaying Markdown in a Jupyter Notebook or on a webpage, we need to convert it to HTML. For most Markdown packages, they will convert two or more consecutive '\n' to a new `<p>`. The example above is a very common use case when comparing an original email to a proofread and corrected email. The current code cannot handle it properly.

Because diff.output_markdown include code like
```
<del>weekend\n\nSophia \n</del><ins>weekend.\n\n\n\nSophia.</ins>'
```
The markdown package will add a `</p><p>` to replace the `\n\n` between `<del></del>` and `<ins></ins>`
The `<del></del>` or `<ins></ins>` will be broken.

The `html=markdown.markdown(diff.output_markdown)` result looks like this.

<p>Happy Saturday,</p><p>Thank you for reaching <del>out, have </del><ins>out. Have </ins>a good <del>weekend</p><p>Sophia </del><ins>weekend.</p><p>Sophia.</ins></p>

You see two Sophia because `<del>weekend\n\nSophia \n</del>` is broken by a `<p>`.

### Solution:
Break the text into paragraphs. For each paragraph, which is now guaranteed to not include '\n', we generate the output_markdown and concatenate them. Add `\n\n` between paragraphs.

The new html result looks like this.

<p>Happy Saturday,</p><p>Thank you for reaching <del>out, have </del><ins>out. Have </ins>a good <del>weekend</del><ins>weekend.</ins></p><p><del>Sophia </del><ins>Sophia.</ins></p>

https://github.com/houfu/redlines/pull/4

**Comment by inteli5** [comment]
There may be a better solution.

**Comment by inteli5** [comment]
Implemented the new solution.

**Comment by houfu** [comment]
Hey, thanks for the contribution! You've definitely found a blind spot in the current code.

I'm wondering whether we should treat new paragraphs as a "token" instead of splitting paragraphs. This is how I imagine Microsoft Word would deal with this issue (it looks like a red "¶" in track changes). Of course, we don't need to follow Microsoft Word, and it's not obvious to me how it would be displayed in markdown for example. What do you think?

I am gonna give this a bit more thought before deciding on one method or another.

**Comment by inteli5** [comment]
Hi, houfu, I have a new idea this morning. How about we replace all the "one or more consecutive `\n`" to "¶", then let SequenceMatcher.get_opcodes() do the work.  Then, when we covert the opcodes to markdown.
```python
for tag, i1, i2, j1, j2 in opcode:
    if tag == "equal":
        result.append("".join(seq1[i1:i2]))
        # and replace "¶" in seq1[i1:i2] to '\n\n'
    elif tag == "insert":
       # split the string by "¶" 
       for split in splits:
             result.append(
             f"<{md_styles['ins'][0]}>{split}</{md_styles['ins'][1]}>")
             result.append("\n\n"
       result. pop() # pop the last '\n\n'
    elif tag == "delete":
        result.append(
            f"<{md_styles['del'][0]}>{''.join(seq1[i1:i2])}</{md_styles['del'][1]}>"
        )
        # no change. 
    elif tag == "replace":
        result.append(
            f"<{md_styles['del'][0]}>{''.join(seq1[i1:i2])}</{md_styles['del'][1]}>"
        )
       # split seq1[j1:j2] by "¶" 
       for split in splits:
             result.append(
             f"<{md_styles['ins'][0]}>{split}</{md_styles['ins'][1]}>")
             result.append("\n\n"
       result. pop() # pop the last '\n\n'
```

**Comment by inteli5** [comment]
I have implemented the algorithm mentioned above, and the code in the current pull request reflects this change.

**Comment by houfu** [comment]
LGTM!

**Comment by houfu** [comment]
Will need some time to tidy up the repo for the new release. You will see your code then.
